### PR TITLE
Fix compilation errors with GCC 7

### PR DESCRIPTION
**Pull request by jlecoeur**
See associated changes in https://github.com/PX4/Firmware/pull/7332

### CODE REVIEW

File: EKF/mag_fusion.cpp [hunk 1]
```diff
@@ -149,9 +149,9 @@ void Ekf::fuseMag()
 		_mag_test_ratio[index] = sq(_mag_innov[index]) / (sq(math::max(_params.mag_innov_gate, 1.0f)) * _mag_innov_var[index]);
 		if (_mag_test_ratio[index] > 1.0f) {
 			healthy = false;
-			_innov_check_fail_status.value |= (1 << (index + 3));
+			_innov_check_fail_status.value |= ((1 << (index + 3)) != 0);
 		} else {
-			_innov_check_fail_status.value &= !(1 << (index + 3));
+			_innov_check_fail_status.value &= !((1 << (index + 3)) != 0);
```
Comment (bkueng) [inline EKF/mag_fusion.cpp]: That does not look correct. I think it should be `&= ~(1 << (index + 3));`

Comment (jlecoeur) [inline EKF/mag_fusion.cpp]: You are right @bkueng ! 
I thought `_innov_check_fail_status` contained only booleans (hence the !=0 test), but `value` is an uint16_t.
I fixed it.

**Comment by LorenzMeier** [comment]
@priseborough This looks like it could be a real issue and should be propagated.

**Comment by CarlOlsson** [comment]
@LorenzMeier 
Yes this was a real issue.
Before this PR
image

After this PR
image

**Pull request merged**